### DISCUSSION TITLE
Performance with multiple GeoJSONLayer

### DISCUSSION THREAD

**Discussion by TextTL** [visgl/deck.gl#8568] [Q&A]
To hide and show features, I'm using multiple GeoJSONLayer. I noticed that if I render the same set of features on a single GeoJSONLayer, zoom and pan are fast. When I use the same set of data on 3 or more GeoJSONLayers, zoom an pan come slugish. The data set is not big, it's about 5000 features with LineString,Polygon, Point (Circle+Icon+text). 

If I used one GeoJSONLayer, and other layers such as PolygonLayer, PathLayer, TextLayer, IconLayer, I did not see performance issue with the dataset.

Is that expected? I'm using the  deckgl 8.9.34.
Thanks,

**Comment by donmccurdy** [comment]
Each layer must draw the data again, so three layers does roughly triple the rendering cost. 5000 features may or may not be a lot, depending on the complexity of the geometry within them. If the performance you're seeing doesn't seem consistent with that explanation, it is always possible there could be a bug, but a demo would be needed to investigate further.
